--- a/cmake_modules/morse_cmake/modules/find/FindSTARPU.cmake	2020-05-02 02:18:41.235505173 -0000
+++ b/cmake_modules/morse_cmake/modules/find/FindSTARPU.cmake	2020-05-02 02:18:43.952589208 -0000
@@ -167,7 +167,7 @@
     mark_as_advanced(CUDA_SDK_ROOT_DIR)
     mark_as_advanced(CUDA_TOOLKIT_ROOT_DIR)
     mark_as_advanced(CUDA_VERBOSE_BUILD)
-    libraries_absolute_path(CUDA_LIBRARIES "$ENV{CUDA_ROOT}/lib64")
+    #libraries_absolute_path(CUDA_LIBRARIES "$ENV{CUDA_ROOT}/lib64")
   endif()
 endif()
 
