--- a/CMakeLists.txt	2020-04-23 02:43:31.104097672 -0000
+++ b/CMakeLists.txt	2020-04-23 02:43:40.338380899 -0000
@@ -865,8 +865,8 @@
 
 install(TARGETS pastix
   RUNTIME DESTINATION bin
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib)
+  ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${INSTALL_LIBDIR})
 
 # Install header files
 set(PASTIX_HDRS
--- a/spm/CMakeLists.txt	2020-04-23 02:44:15.458458074 -0000
+++ b/spm/CMakeLists.txt	2020-04-23 02:44:21.571645575 -0000
@@ -238,8 +238,8 @@
 endif (MPI_C_FOUND)
 
 install(TARGETS spm
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib)
+  ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${INSTALL_LIBDIR})
 
 install(FILES
   include/spm.h
--- a/cmake_modules/morse_cmake/modules/GenPkgConfig.cmake	2020-04-23 02:53:32.923577344 -0000
+++ b/cmake_modules/morse_cmake/modules/GenPkgConfig.cmake	2020-04-23 02:53:39.578781790 -0000
@@ -196,7 +196,7 @@
     # ------------
     install(FILES
       "${CMAKE_BINARY_DIR}/lib/pkgconfig/${fname}.pc"
-      DESTINATION lib/pkgconfig )
+      DESTINATION ${INSTALL_LIBDIR}/pkgconfig )
 
   endforeach()
 
--- a/kernels/gpus/CMakeLists.txt	2020-04-23 02:52:27.691573447 -0000
+++ b/kernels/gpus/CMakeLists.txt	2020-04-23 02:52:38.915918254 -0000
@@ -66,8 +66,8 @@
 #     ${cuda_generated_files})
 #   cuda_add_cublas_to_target( pastix_cucores_sm${_smtarget} )
 #   install(TARGETS pastix_cucores_sm${_smtarget}
-#     LIBRARY DESTINATION lib
-#     ARCHIVE DESTINATION lib )
+#     LIBRARY DESTINATION ${INSTALL_LIBDIR}
+#     ARCHIVE DESTINATION ${INSTALL_LIBDIR} )
 #   # if( NOT CMAKE_BUILD_SHARED_LIBS )
 #   #   target_link_libraries( pastix_cucores pastix_cucores_sm${_smtarget} )
 #   # endif()
@@ -79,7 +79,7 @@
 cuda_add_cublas_to_target( pastix_kernels_cuda )
 
 install(TARGETS pastix_kernels_cuda
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib )
+  LIBRARY DESTINATION ${INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${INSTALL_LIBDIR} )
 
 cuda_build_clean_target()
--- a/kernels/CMakeLists.txt	2020-04-23 02:53:09.951871665 -0000
+++ b/kernels/CMakeLists.txt	2020-04-23 02:53:13.544982040 -0000
@@ -153,8 +153,8 @@
 
 ### Install library
 install(TARGETS pastix_kernels
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib)
+  ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${INSTALL_LIBDIR})
 
 ### Add generated files to the global property
 add_documented_files(
--- a/sopalin/parsec/CMakeLists.txt	2020-04-23 02:51:56.303609217 -0000
+++ b/sopalin/parsec/CMakeLists.txt	2020-04-23 02:52:14.449166644 -0000
@@ -114,8 +114,8 @@
 endif (MPI_C_FOUND)
 
 install(TARGETS pastix_parsec
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib)
+  ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${INSTALL_LIBDIR})
 
 ### Add non-generated documented files to the global property
 add_documented_files(
--- a/sopalin/starpu/CMakeLists.txt	2020-04-23 02:50:53.436677975 -0000
+++ b/sopalin/starpu/CMakeLists.txt	2020-04-23 02:51:11.073219764 -0000
@@ -96,8 +96,8 @@
 endif (MPI_C_FOUND)
 
 install(TARGETS pastix_starpu
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib)
+  ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${INSTALL_LIBDIR})
 
 ### Add non-generated documented files to the global property
 add_documented_files(
--- a/wrappers/fortran90/CMakeLists.txt	2020-04-23 02:51:36.207991890 -0000
+++ b/wrappers/fortran90/CMakeLists.txt	2020-04-23 02:51:45.424275012 -0000
@@ -31,8 +31,8 @@
 target_link_libraries( pastixf pastix )
 install(TARGETS pastixf
   RUNTIME DESTINATION bin
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib )
+  ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${INSTALL_LIBDIR} )
 
 #
 # Add examples
--- a/wrappers/python/CMakeLists.txt	2020-04-23 02:49:36.115302694 -0000
+++ b/wrappers/python/CMakeLists.txt	2020-04-23 02:50:05.881217089 -0000
@@ -27,7 +27,7 @@
   ${CMAKE_CURRENT_SOURCE_DIR}/examples/pypastix/pastix.py
   ${CMAKE_CURRENT_SOURCE_DIR}/examples/pypastix/enum.py
   ${CMAKE_CURRENT_SOURCE_DIR}/examples/pypastix/solver.py
-  DESTINATION lib/python/pypastix )
+  DESTINATION ${PYTHON_SITE_PACKAGES_DIR}/pypastix )
 
 # Install python examples
 install(FILES
--- a/spm/cmake_modules/morse_cmake/modules/GenPkgConfig.cmake	2020-04-23 02:46:36.441783200 -0000
+++ b/spm/cmake_modules/morse_cmake/modules/GenPkgConfig.cmake	2020-04-23 02:46:51.652250457 -0000
@@ -196,7 +196,7 @@
     # ------------
     install(FILES
       "${CMAKE_BINARY_DIR}/lib/pkgconfig/${fname}.pc"
-      DESTINATION lib/pkgconfig )
+      DESTINATION ${INSTALL_LIBDIR}/pkgconfig )
 
   endforeach()
 
--- a/spm/wrappers/fortran90/CMakeLists.txt	2020-04-23 02:47:11.914872915 -0000
+++ b/spm/wrappers/fortran90/CMakeLists.txt	2020-04-23 02:47:17.813054107 -0000
@@ -31,8 +31,8 @@
 target_link_libraries( spmf spm )
 install(TARGETS spmf
   RUNTIME DESTINATION bin
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib )
+  ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${INSTALL_LIBDIR} )
 
 #
 # Add examples
--- a/spm/wrappers/python/CMakeLists.txt	2020-04-23 02:47:31.661479524 -0000
+++ b/spm/wrappers/python/CMakeLists.txt	2020-04-23 02:49:02.776278533 -0000
@@ -26,7 +26,7 @@
   ${CMAKE_CURRENT_SOURCE_DIR}/spm/__spm__.py
   ${CMAKE_CURRENT_SOURCE_DIR}/spm/spm.py
   ${CMAKE_CURRENT_SOURCE_DIR}/spm/enum.py
-  DESTINATION lib/python/spm )
+  DESTINATION ${PYTHON_SITE_PACKAGES_DIR}/spm )
 
 # Install python examples
 install(FILES
