--- a/src/ClpCholeskyMumps.cpp	2020-04-05 18:43:53.660150535 -0000
+++ b/src/ClpCholeskyMumps.cpp	2020-04-05 18:44:06.323541158 -0000
@@ -13,6 +13,7 @@
 #define USE_COMM_WORLD -987654
 extern "C" {
 #include "dmumps_c.h"
+}
 // In newer ThirdParty/Mumps, mpi.h is renamed to mumps_mpi.h.
 // We get informed about this by having COIN_USE_MUMPS_MPI_H defined.
 #ifdef COIN_USE_MUMPS_MPI_H
@@ -20,7 +21,6 @@
 #else
 #include "mpi.h"
 #endif
-}
 
 #include "ClpCholeskyMumps.hpp"
 #include "ClpMessage.hpp"
