--- a/SRC/CMakeLists.txt	2020-04-25 07:03:30.396708562 -0000
+++ b/SRC/CMakeLists.txt	2020-04-25 07:03:38.666963787 -0000
@@ -158,7 +158,7 @@
   list(APPEND targets superlu_dist-static)
 endif()
 
-set(superlu_dist_libs ${MPI_C_LIBRARIES} ${BLAS_LIB} ${LAPACK_LIB}
+set(superlu_dist_libs ${MPI_C_LIBRARIES} ${MPI_CXX_LIBRARIES} ${BLAS_LIB} ${LAPACK_LIB}
     ${PARMETIS_LIB} ${COMBBLAS_LIB})
 
 if (NOT MSVC)
