--- a/CMakeLists.txt	2020-09-20 18:52:30.487372271 -0000
+++ b/CMakeLists.txt	2020-09-20 18:53:52.463369420 -0000
@@ -606,7 +606,7 @@
 # bench, demo and tests
 
 set(COPY_DEMO_TEST_DEMO_DATA FALSE)
-if (NOT EXISTS ${DOLFIN_SOURCE_DIR}/demo/documented/poisson/cpp/Poisson.h)
+if (NOT "${CROSS_COMPILING}" AND NOT EXISTS ${DOLFIN_SOURCE_DIR}/demo/documented/poisson/cpp/Poisson.h)
   message(STATUS "")
   message(STATUS "Generating form files in demo, test and bench directories. May take some time...")
   message(STATUS "----------------------------------------------------------------------------------------")
