--- a/config/PETSc/options/installDir.py	2020-04-03 20:49:43.841857231 -0000
+++ b/config/PETSc/options/installDir.py	2020-04-03 20:49:54.934199012 -0000
@@ -41,8 +41,7 @@
       self.petscDir = self.dir
       self.petscArch = ''
       try:
-        os.makedirs(os.path.join(self.dir,'PETScTestDirectory'))
-        os.rmdir(os.path.join(self.dir,'PETScTestDirectory'))
+        pass
       except Exception as e:
         self.logPrint('Error trying to to test write permissions on directory '+str(e))
         self.installSudoMessage = 'You do not have write permissions to the --prefix directory '+self.dir+'\nYou will be prompted for the sudo password for any external package installs'
