--- a/CMakeLists.txt.orig2	2020-04-20 21:46:53.459065474 -0000
+++ b/CMakeLists.txt	2020-04-20 21:47:00.234273462 -0000
@@ -39,6 +39,7 @@
 endif()
 
 set(CMAKE_Fortran_MODULE_DIRECTORY ${LIBSUPERMESH_BINARY_DIR}/include)
+include_directories(${LIBSUPERMESH_SOURCE_DIR})
 include_directories(${LIBSUPERMESH_SOURCE_DIR}/include)
 include_directories(${LIBSUPERMESH_BINARY_DIR}/include)
 include_directories(${LIBSUPERMESH_BINARY_DIR}/include_local)
