--- a/pyop2/compilation.py	2020-10-15 16:53:45.000000000 -0000
+++ b/pyop2/compilation.py	2020-10-15 16:58:53.000000000 -0000
@@ -69,13 +69,13 @@
     except (subprocess.CalledProcessError, UnicodeDecodeError):
         return CompilerInfo("unknown", version.LooseVersion("unknown"))
 
-    if ver.startswith("gcc"):
+    if "gcc" in ver:
         compiler = "gcc"
-    elif ver.startswith("clang"):
+    elif "clang" in ver:
         compiler = "clang"
-    elif ver.startswith("Apple LLVM"):
+    elif "Apple LLVM" in ver:
         compiler = "clang"
-    elif ver.startswith("icc"):
+    elif "icc" in ver:
         compiler = "icc"
     else:
         compiler = "unknown"
