--- a/mlir/cmake/modules/AddMLIR.cmake	2020-06-24 18:22:47.443891626 -0000
+++ b/mlir/cmake/modules/AddMLIR.cmake	2020-06-24 18:22:54.451109754 -0000
@@ -141,6 +141,7 @@
   set(export_to_mlirtargets)
   if (${name} IN_LIST LLVM_DISTRIBUTION_COMPONENTS OR
       "mlir-libraries" IN_LIST LLVM_DISTRIBUTION_COMPONENTS OR
+      "mlir" IN_LIST LLVM_ENABLE_PROJECTS OR
       NOT LLVM_DISTRIBUTION_COMPONENTS)
       set(export_to_mlirtargets EXPORT MLIRTargets)
     set_property(GLOBAL PROPERTY MLIR_HAS_EXPORTS True)
