--- a/firedrake/checkpointing.py	2020-10-15 01:14:41.000000000 -0000
+++ b/firedrake/checkpointing.py	2020-10-15 01:14:46.000000000 -0000
@@ -1,10 +1,10 @@
 from firedrake.petsc import PETSc
 from pyop2.mpi import COMM_WORLD, dup_comm, free_comm
-from firedrake.cython import hdf5interface as h5i
+#from firedrake.cython import hdf5interface as h5i
 import firedrake
 import numpy as np
 import os
-import h5py
+#import h5py
 
 
 __all__ = ["DumbCheckpoint", "HDF5File", "FILE_READ", "FILE_CREATE", "FILE_UPDATE"]
