--- a/firedrake/paraview_reordering.py	2020-04-22 00:52:12.351319088 -0000
+++ b/firedrake/paraview_reordering.py	2020-04-22 00:52:16.527447627 -0000
@@ -1,23 +1,10 @@
 from tsfc.fiatinterface import create_element
 import numpy as np
 from pyop2.utils import as_tuple
-import importlib
-"""
-This requires an explentation.
-Vtk has some .so deps that might not be present (e.g. libsm.so (X11 Sessions))
-However, we only need vtkCommonKitPython, which, according to ldd, only cares about
-things that we should expect: libc, libdl.so, libstdc++, libm, libgcc_s.
-Thus, we hackily import the module that lives in vtkCommonKitPython.so
-"""
-vtkSoLoc = importlib.util.find_spec("vtk").submodule_search_locations[0]
-vtkSoLoc += "/vtkCommonKitPython.so"
-loader = importlib.machinery.ExtensionFileLoader("vtkCommonKitPython", vtkSoLoc)
-mod = loader.load_module("vtkCommonKitPython")
-vtkLagrangeTetra = mod.vtkLagrangeTetra
-vtkLagrangeHexahedron = mod.vtkLagrangeHexahedron
-vtkLagrangeTriangle = mod.vtkLagrangeTriangle
-vtkLagrangeQuadrilateral = mod.vtkLagrangeQuadrilateral
-vtkLagrangeWedge = mod.vtkLagrangeWedge
+
+from vtk.vtkCommonDataModelPython import \
+        vtkLagrangeTetra, vtkLagrangeHexahedron, vtkLagrangeTriangle, \
+        vtkLagrangeQuadrilateral, vtkLagrangeWedge
 
 
 def firedrake_local_to_cart(element):
