--- a/configure.ac	2020-04-29 21:38:31.697260849 -0000
+++ b/configure.ac	2020-04-29 21:41:12.336223570 -0000
@@ -2231,7 +2231,14 @@
 
 ##echo @@@@@@@@@@@@ $ff_petsc_ok @@ "hpddm.$DYLIB_SUFFIX $FF_PETCS_DYLIB"
 TOOL_DISABLE(hpddm,["hpddm.$DYLIB_SUFFIX hpddm_substructuring.$DYLIB_SUFFIX $FF_PETSC_DYLIB"])
-TOOL_DISABLE(bem,["bem.$DYLIB_SUFFIX"])
+TOOL_DISABLE(bemtool,["bemtool.$DYLIB_SUFFIX"])
+
+# Catch old usage to help users update
+AC_ARG_ENABLE(bem, AS_HELP_STRING([--enable-bem],
+  ["DEPRECATED: use --enable-bemtool instead"]))
+if test -n "$enable_bem"; then
+  AC_MSG_ERROR(["Arguments --enable-bem/--disable-bem are deprecated: use --enable-bemtool/--disable-bemtool instead"])
+fi
 
 ##
 if test "$ff_mkl_ok" = yes ; then
