--- a/configure.ac	2020-04-29 22:57:53.036651591 -0000
+++ b/configure.ac	2020-04-29 23:21:45.033825942 -0000
@@ -58,7 +58,7 @@
 	    ff_$1_ok=yes
 	    AC_MSG_NOTICE([    ++ add $1 : $2 ;  $3 in  $ff_where_lib_conf "])
 	else
-	    AC_MSG_NOTICE([    -- do not add $1 : $2 $3 in  $ff_where_lib_conf "])
+	    AC_MSG_NOTICE([    -- already added $1 : $2 $3 in  $ff_where_lib_conf "])
         fi
 ]) dnl
 m4_define([AC_FF_WHERELIB],
@@ -123,7 +123,7 @@
 
 	# some directives have been specified, use them instead of downloading
 
-	AC_FF_ADDWHERELIB($1,$with_$1_ldflags,$with_$1_include)
+	AC_FF_WHERELIB($1,$with_$1_ldflags,$with_$1_include)
         AC_SUBST([TOOL_COMPILE_$1],"")
 	enable_$1_download=no
     fi
@@ -2026,32 +2026,6 @@
   AC_FF_ADDWHERELIB(gsl,$GSL_LIBS,$GSL_CFLAGS)
 fi
 
-m4_map([AC_FF_WHERELIB],[
-	[[mumps],[-ldmumps -lzmumps  -lmumps_common  -lpord],[dmumps_c.h],[]],
-	[[mumps_seq],[-ldmumps_seq -lzmumps_seq  -lmumps_common_seq  -lpord_seq -lmpiseq_seq],[dmumps_c.h],[]],
-	[[libseq],[-lmpiseq_seq],[${inc_usr_include}mumps_seq/mpi.h]],
-	[[mumps_ptscotch],[-lpord_ptscotch -lmumps_common_ptscotch -ldmumps_ptscotch -lzmumps_ptscotch -lpord_ptscotch],[dmumps_c.h]],
-	[[mumps_scotch],[-lpord_scotch -lmumps_common_scotch -ldmumps_scotch -lzmumps_scotch -lpord_scotch],[dmumps_c.h]],
-	[[fftw3],[-lfftw3],[${inc_usr_include}fftw3.h],[]],
-	[[superlu],[-lsuperlu],[${inc_usr_include}superlu/superlu_enum_consts.h],[]],
-        [[blacs],[ -lblacsCinit$ff_with_mpi	-lblacsF77init$ff_with_mpi	-lblacs$ff_with_mpi],[]],
-        [[scalapack],[-lscalapack$ff_with_mpi],[]],
-        [[scotch],[-lscotch -lscotcherr],[scotch.h]],
-        [[ptscotch],[-lptscotch -lptscotcherr],[ptscotch.h]],
-	[[metis],[-lmetis],[${inc_usr_include}metis/metis.h],[]],
-	[[metis],[-lmetis],[metis.h],[]],
-	[[parmetis],[-lparmetis -lmetis],[],[]],
-	[[freeyams],[-lfreeyams],[freeyamslib.h],[]],
-	[[mmg3d],[-lmmg3d-v4],[libmmg3d.h],[]],
-        [[mmg],[-lmmg],[libmmg.h],[]],
-        [[parmmg],[-lparmmg],[libparmmg.h],[]],
-	[[mshmet],[-lmshmet],[mshmetlib.h],[]],
-dnl	[[gsl],[-lgsl -lgslcblas -lm],[gsl/gsl_sf.h],[]],
-	[[tetgen],[-ltet],[tetgen.h],[]],
-	[[ipopt],[ -lipopt],[${inc_usr_include}coin/IpTNLP.hpp],[]],
-	[[nlopt],[ -lnlopt],[nlopt.hpp],[]]
-    ]
-    )
 ## before try
 
 TOOL_DISABLE(tetgen,tetgen.$DYLIB_SUFFIX)
@@ -2079,6 +2053,36 @@
 AC_SUBST([BIN_ffmaster],"$BIN_ffmaster")
 TOOL_DISABLE(MMAP,ff-mmap-semaphore.$DYLIB_SUFFIX,,nodownload)
 
+# Libraries for which a --with-X-{include,ldflags} argument was given,
+# were already checked with AC_FF_WHERELIB within the TOOL_DISABLE calls
+# above; the remaining libraries without the arguments will be checked
+# with the default arguments specified below.
+m4_map([AC_FF_WHERELIB],[
+    [[mumps],[-ldmumps -lzmumps  -lmumps_common  -lpord],[dmumps_c.h],[]],
+    [[mumps_seq],[-ldmumps_seq -lzmumps_seq  -lmumps_common_seq  -lpord_seq -lmpiseq_seq],[dmumps_c.h],[]],
+    [[libseq],[-lmpiseq_seq],[${inc_usr_include}mumps_seq/mpi.h]],
+    [[mumps_ptscotch],[-lpord_ptscotch -lmumps_common_ptscotch -ldmumps_ptscotch -lzmumps_ptscotch -lpord_ptscotch],[dmumps_c.h]],
+    [[mumps_scotch],[-lpord_scotch -lmumps_common_scotch -ldmumps_scotch -lzmumps_scotch -lpord_scotch],[dmumps_c.h]],
+    [[fftw3],[-lfftw3],[${inc_usr_include}fftw3.h],[]],
+    [[superlu],[-lsuperlu],[${inc_usr_include}superlu/superlu_enum_consts.h],[]],
+    [[blacs],[ -lblacsCinit$ff_with_mpi    -lblacsF77init$ff_with_mpi    -lblacs$ff_with_mpi],[]],
+    [[scalapack],[-lscalapack$ff_with_mpi],[]],
+    [[scotch],[-lscotch -lscotcherr],[scotch.h]],
+    [[ptscotch],[-lptscotch -lptscotcherr],[ptscotch.h]],
+    [[metis],[-lmetis],[${inc_usr_include}metis/metis.h],[]],
+    [[metis],[-lmetis],[metis.h],[]],
+    [[parmetis],[-lparmetis -lmetis],[],[]],
+    [[freeyams],[-lfreeyams],[freeyamslib.h],[]],
+    [[mmg3d],[-lmmg3d-v4],[libmmg3d.h],[]],
+    [[mmg],[-lmmg],[libmmg.h],[]],
+    [[parmmg],[-lparmmg],[libparmmg.h],[]],
+    [[mshmet],[-lmshmet],[mshmetlib.h],[]],
+dnl    [[gsl],[-lgsl -lgslcblas -lm],[gsl/gsl_sf.h],[]],
+    [[tetgen],[-ltet],[tetgen.h],[]],
+    [[ipopt],[ -lipopt],[${inc_usr_include}coin/IpTNLP.hpp],[]],
+    [[nlopt],[ -lnlopt],[nlopt.hpp],[]]
+])
+
 # FFCS - MUMPS_seq has a different Win32 compiler setup from FFCS, so we need to add some extra parameters
 
 if test "$OS" = Windows_NT
