--- a/configure.ac	2020-04-29 20:39:28.649776101 -0000
+++ b/configure.ac	2020-04-29 20:39:30.523833994 -0000
@@ -800,7 +800,8 @@
 ff_libs="$LIBS"
 AC_ARG_WITH(flib,[  --with-flib= the fortran library ])
 AC_ARG_ENABLE([opengl],AS_HELP_STRING([--enable-opengl],
-                                      [Enable/disable support for OpenGL]))
+                                      [Enable/disable support for OpenGL]),
+				      enable_opengl=no)
 
 # echo "  .... LIBS = $LIBS .... "
 if test x$enable_opengl != xno; then
