--- a/CMakeLists.txt	2020-09-20 17:23:03.867558871 -0000
+++ b/CMakeLists.txt	2020-09-20 17:33:09.059537828 -0000
@@ -143,15 +143,17 @@
 SET( LIBRARIES ${MPI_C_LIBRARIES} ${LIBRARIES} )
 SET( LIBRARIES ${MPI_CXX_LIBRARIES} ${LIBRARIES} )
 
-EXECUTE_PROCESS ( COMMAND  ${MPIEXEC} --help mapping
-  RESULT_VARIABLE CMD_ERROR
-  OUTPUT_VARIABLE CMD_OUTPUT )
+IF (NOT "${CROSS_COMPILING}")
+  EXECUTE_PROCESS ( COMMAND  ${MPIEXEC} --help mapping
+    RESULT_VARIABLE CMD_ERROR
+    OUTPUT_VARIABLE CMD_OUTPUT )
 
-STRING(REGEX REPLACE "\"" " " CMD_OUT ${CMD_OUTPUT})
+  STRING(REGEX REPLACE "\"" " " CMD_OUT ${CMD_OUTPUT})
 
-IF ( ${CMD_OUT} MATCHES "oversubscribe" )
-  MESSAGE ( STATUS "OK" )
-  SET ( MPI_ARGS "-oversubscribe" )
+  IF ( ${CMD_OUT} MATCHES "oversubscribe" )
+    MESSAGE ( STATUS "OK" )
+    SET ( MPI_ARGS "-oversubscribe" )
+  ENDIF()
 ENDIF()
 
 ###############################################################################
