--- a/3rd-party/prrte/config/prte_check_alps.m4	2020-10-03 00:50:39.901388826 -0000
+++ b/3rd-party/prrte/config/prte_check_alps.m4	2020-10-03 00:51:20.821390889 -0000
@@ -92,8 +92,10 @@
                     [AC_HELP_STRING([--with-alps(=DIR|yes|no)],
                     [Build with ALPS scheduler component, optionally adding DIR/include, DIR/lib, and DIR/lib64 to the search path for headers and libraries (default: auto)])],[],with_alps=auto)
 
-        if test -f /etc/opt/cray/release/clerelease; then
-            cle_level=`awk -F. '{print [$]1}' /etc/opt/cray/release/clerelease`
+        if test -f /etc/opt/cray/release/cle-release; then
+            cle_level=`$AWK '/^RELEASE=/{print gensub(/^RELEASE=([[0-9]]+).*/, "\\\\1", "g")}' /etc/opt/cray/release/cle-release`
+        elif test -f /etc/opt/cray/release/clerelease; then
+            cle_level=`$AWK -F. '{print [$]1}' /etc/opt/cray/release/clerelease`
         else
             cle_level="unknown"
         fi
